scripts/check-file-changes.sh

```diff
@@ -1,7 +1,11 @@
 #!/bin/bash
+# To prevent the tests/builds to run for only a doc change, this script checks what files have changed in a pull request.
 
 BRANCH="$(git rev-parse --abbrev-ref HEAD)"
 echo $BRANCH
-if [[ "$BRANCH" != "master" ]]; then
-    git diff master --name-only --no-color | egrep -e "^(\.circleci\/.*)|(.*\.(go|mod|sum))|Makefile$" || circleci step halt;
+if [[ "$BRANCH" != "master" ]] && [[ "$BRANCH" != release* ]]; then # This should never skip for master and release branches
+    # Ask git for all the differences between this branch and master
+    # Then use grep to look for changes in the .circleci/ directory, anything named *.go* or *.mod* or *.sum* or *.sh* or *Makefile
```

Comment:
Nitpick to change the comments to match the regexp now that it has ^$ in each alternate section:
```suggestion
    # Then use grep to look for changes in the .circleci/ directory, anything named *.go or *.mod or *.sum or *.sh or Makefile
```